cpp/src/arrow/acero/aggregate_node.cc

```diff
@@ -398,25 +398,25 @@ class ScalarAggregateNode : public ExecNode, public TracedNode {
     auto aggregates = aggregate_options.aggregates;
     const auto& keys = aggregate_options.keys;
     const auto& segment_keys = aggregate_options.segment_keys;
-    const auto concurreny =
-        plan->query_context()->exec_context()->executor()->GetCapacity();
+    const auto concurrency = plan->query_context()->max_concurrency();
+    // We can't use concurrency == 1 because that include I/O concurrency
```

Comment:
Do you have an issue with this code?
https://github.com/apache/arrow/blob/be12888997c81b1fb7947f6284be1256edd4d3e4/cpp/src/arrow/acero/hash_join_node.cc#L929

Reply:
I don't think so.  You'll notice, in the very next line, it is doing the same calculation that max_concurrency does.

Reply:
I see. Thanks!